packages/resource-deployment/templates/dashboard.template.json

```diff
@@ -1103,7 +1103,7 @@
                                         },
                                         {
                                             "name": "Query",
-                                            "value": "let startTime = startofmonth(now());\nlet timeRangeInSeconds = (now() - startTime) / 1s;\nlet failureOrRecoverEvents = availabilityResults \n| serialize \n| extend prevSuccess = prev(success)\n| where timestamp > startTime and name == 'foo' and (success != prevSuccess)\n| sort by timestamp asc  \n| project timestamp , name, success, prevSuccess;\nlet availableIntervals = failureOrRecoverEvents\n| serialize \n| where success == \"1\" and prevSuccess == \"0\" \n| extend prevTimeStamp = prev(timestamp), status=\"available\"\n| extend intervalInSeconds = todouble((timestamp - prevTimeStamp)/1s) \n| extend totoalAvailableTime = row_cumsum(intervalInSeconds);\navailableIntervals\n| sort by timestamp desc\n| take 1\n| extend  availability = strcat(tostring(round(totoalAvailableTime/timeRangeInSeconds * 100, 2)), \"%\"), AST=timeRangeInSeconds\n| project name, availability, totoalAvailableTime, AST\n"
+                                            "value": "let startTime = ago(30d);\nlet timeRangeInMinutes = (now() - startTime) / 1min;\nlet failureOrRecoverEvents = availabilityResults\n| serialize \n| sort by timestamp asc  \n| where timestamp > startTime\n| extend prevSuccess = prev(success, 1, \"0\"), nextSuccess=next(success, 1, \"0\") \n| where name == 'workerAvailabilityTest' and (success != prevSuccess or success != nextSuccess)\n| project timestamp , name, success, prevSuccess, nextSuccess;\nlet availableIntervals = failureOrRecoverEvents\n| serialize \n| sort by timestamp asc \n| extend start = timestamp, end=next(timestamp, 1, now())\n| where success == \"1\"\n| extend interval_in_minutes = todouble((end - start)/1min)\n| extend totoalAvailableTime_in_minutes = row_cumsum(interval_in_minutes);\navailableIntervals\n| sort by timestamp desc\n| take 1\n| extend  availability = strcat(tostring(round(totoalAvailableTime_in_minutes/timeRangeInMinutes * 100, 2)), \"%\"), AST_in_minutes=timeRangeInMinutes\n| project availability, totoalAvailableTime_in_minutes, AST_in_minutes\n"
```

Comment:
can this 30d ago be not hardcoded here? it should be set at the top in the dashboard.

Reply:
you should be able to change the time frame & granularity